kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1288 @@
+        TSystemVectorPointerType& pDx,
+        TSystemVectorPointerType& pb,
+        ModelPart& rModelPart
+        ) override
+    {
+        // We resize the basic system
+        BaseType::ResizeAndInitializeVectors(pScheme, pA, pDx, pb, rModelPart);
+
+        // Now we resize the constraint system
+        if(rModelPart.MasterSlaveConstraints().size() > 0) {
+            if (mpTMatrix == NULL) { // If the pointer is not initialized initialize it to an empty matrix
+                TSystemMatrixPointerType pNewT = TSystemMatrixPointerType(new TSystemMatrixType(0, 0));
+                mpTMatrix.swap(pNewT);
+            }
+
+            TSystemMatrixType& rTMatrix = *mpTMatrix;
+
+            // Resizing the system vectors and matrix
+            if (rTMatrix.size1() == 0 || BaseType::GetReshapeMatrixFlag() || mCleared) { // If the matrix is not initialized
+                rTMatrix.resize(BaseType::mEquationSystemSize, mDoFToSolveSystemSize, false);
+                ConstructRelationMatrixStructure(pScheme, rTMatrix, rModelPart);
+            } else {
+                if (rTMatrix.size1() != BaseType::mEquationSystemSize || rTMatrix.size2() != mDoFToSolveSystemSize) {
+                    KRATOS_ERROR <<"The equation system size has changed during the simulation. This is not permited."<<std::endl;
+                    rTMatrix.resize(BaseType::mEquationSystemSize, mDoFToSolveSystemSize, true);
+                    ConstructRelationMatrixStructure(pScheme, rTMatrix, rModelPart);
+                }
+            }
+        }
+    }
+
+    /**
+     * @brief This function is intended to be called at the end of the solution step to clean up memory storage not needed
+     */
+    void Clear() override
+    {
+        BaseType::Clear();
+
+        mDoFToSolveSet = DofsArrayType();
+        mSolvableDoFReorder.clear();
+        mSlaveMasterDoFRelation.clear();
+
+        // Set the flag
+        mCleared = true;
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() > 1) << "Clear Function called" << std::endl;
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    TSystemMatrixPointerType mpTMatrix = NULL; /// This is matrix containing the global relation for the constraints
+
+    DofsArrayType mDoFToSolveSet;              /// The set containing the solvable DoF of the system
+
+    SizeType mDoFToSolveSystemSize = 0;        /// Number of degrees of freedom of the problem to actually be solved
+
+    bool mCleared = true; /// If the system has been reseted
+
+    std::unordered_map<IndexType, IndexType> mSolvableDoFReorder; /// The correlation between the global total DoF order and the solvable DoF order
```

Comment:
I would use a vector for this, to keep the order. the relation is stored in matrix T, so you want to keep the order to be able to use that

what i mean is that you know

       y = T*x

then you want to have a direct mapping between y->slave_dofs x->master_dofs
for this reason the list of dofs on the two side must be ordered

Reply:
Oh this map is still unordered, sorry